pandas/tests/frame/test_constructors.py

```diff
@@ -528,6 +528,29 @@ def test_constructor_dict_of_tuples(self):
         expected = DataFrame({k: list(v) for k, v in data.items()})
         tm.assert_frame_equal(result, expected, check_dtype=False)
 
+    def test_constructor_dict_of_ranges(self):
+        data = {'a': range(3), 'b': range(3, 6)}
+
```

Comment:
This worked in master, but I just added this test as ensurance against problems like in #26342